### ISSUE TITLE
Queue leaks memory through sizeSignal

### ISSUE THREAD

**Issue by lisa-lionheart** [aol/simple-react#92]
So I'm currently using simple-react in a server application that has some long lived queues that process millions of objects a day. For the most part it works great but I have noticed that over time it consumes more and more memory at the rate of about ~8mb an hour.

After some analysis i found that memory is being held through the sizeSignal member of the queue class which has a LinkedBlockingQueue that grows and grows ....

The following code demonstrates the leak:

```
public class QueueTest {
    static class BigObject {
        public final byte[] data = new byte[1_000];
    }
    public static void main(String[] args) {
        Queue<BigObject> queue = new Queue<>(new ArrayBlockingQueue<>(100));
        Thread producer = new Thread(()->{
             while (true){
                queue.add(new BigObject());
            }
        });
        producer.start();
        queue.stream().forEach(o->{
            System.out.println(o.toString());
        });
    }
}
```

Screen shot from visual VM after running for 5 minutes: 
![image](https://cloud.githubusercontent.com/assets/2095125/12404117/c78bf18a-be30-11e5-8105-b14cef9e2265.png)

**Comment by lisa-lionheart** [comment]
![image](https://cloud.githubusercontent.com/assets/2095125/12404228/7d5754f0-be31-11e5-89fa-8bd048b2ae52.png)

**Comment by johnmcclean-aol** [comment]
Thanks for this Lisa, I think the simplest thing to do here is to remove the size signal I suspect it is not used on the vast majority of Queues (possibly we can put a hook into Queue to users can receive size events that they can pass to a user managed signal instead).

**Comment by johnmcclean-aol** [comment]
Fixed in https://github.com/aol/simple-react/pull/91, will be released over the next day or two.

I decided to leave the option of adding a size Signal to the Queue, it's useful for testing properties of the queue, but they aren't included by default.  If users add a size Signal, most likely they will want that data and hence drain it themselves.

**Comment by lisa-lionheart** [comment]
Awesome thanks for the quick response

**Comment by johnmcclean-aol** [comment]
Released 0.99.7 - should be available via maven central shortly.
